### PR TITLE
LOG-1874: lokistack-gateway component restarting too many times on OpenShift

### PR DESCRIPTION
**Pull request by sasagarw**
This PR:
- fixes serviceAccount issue of gateway for static, dynamic modes. Currently, we were setting ServiceAccountName for all the 3 modes but this is not correct as we don't create any ServiceAccount for static and dynamic modes. Hence the deployment errors out saying missing ServiceAccountName and a reconcile error occurs. Hence moved this to openshift specific deployment configuration (this was done in #101)
- degraded condition on the lokistack was getting set but the deployments were still taking places because of the non-nil error returned by `SetDegradedCondition` function.
- fixed the `uuid` and `cookieSecret` issue by fetching the existing tenants.yaml configuration and using the same value in subsequent reconcile. Both uuid and cookieSecret will be created only once.

/cc @periklis 

### Links
- Jira: https://issues.redhat.com/browse/LOG-1874

### CODE REVIEW

**Comment by sasagarw** [comment]
@periklis can we do same with the `cookieSecret` as well?

File: bundle/manifests/loki-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -45,7 +45,7 @@ metadata:
     operatorframework.io/arch.amd64: supported
     operatorframework.io/arch.ppc64le: supported
     operatorframework.io/arch.s390x: supported
-  name: loki-operator.v0.0.1
+  name: loki-operator.v0.1.98
```
Comment (periklis) [inline bundle/manifests/loki-operator.clusterserviceversion.yaml:48]: Can we drop this random version please?

File: bundle/manifests/loki-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -594,8 +594,8 @@ spec:
                   value: quay.io/observatorium/api:latest
                 - name: RELATED_IMAGE_OPA
                   value: quay.io/observatorium/opa-openshift:latest
-                image: quay.io/openshift-logging/loki-operator:v0.0.1
-                imagePullPolicy: IfNotPresent
+                image: quay.io/sasagarw/loki-operator:v0.1.98
+                imagePullPolicy: Always
```
Comment (periklis) [inline bundle/manifests/loki-operator.clusterserviceversion.yaml:598]: Same here please keep the old defaults

File: internal/handlers/internal/gateway/tenant_configmap.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		return nil
+	}
+
+	tcm, err := extractTenantConfigMap(&tenantConfigMap)
+	if err != nil {
+		log.Error(err, "error occurred in extracting tenants.yaml configMap.")
+		return nil
+	}
+
+	tcmMap := make(map[string]map[string]string)
```
Comment (periklis) [inline internal/handlers/internal/gateway/tenant_configmap.go:57]: Suggest to make the inside map a small struct, e.g. `tenantData`

**Review by periklis — CHANGES_REQUESTED** [review]
Looks good, only a couple of small improvements required.

**Review by periklis — APPROVED** [review]

**Pull request merged**